conformance-tests/README.md

```diff
@@ -0,0 +1,54 @@
+# Running [Gateway Conformance Tests](https://gateway-api.sigs.k8s.io/concepts/conformance/#3-conformance-tests) in Kind
```

Comment:
Can you add a link to this doc in our [testing](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/docs/developer/testing.md) doc?